heap_sort/heap_sort.c

```diff
@@ -0,0 +1,63 @@
+/*
+ * C program to Implement Heap sort.
+ * Time complexity: Average = O( n log( n ) ), worst case complexity = O( n log( n ) ).
+ * Space complexity: O(1).
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+// root = index root of the subtree, a is an array, heapsize is size of heap
+void  max_heapify(int a[], int root, int heapsize) {
+    int largest;
+    int l = (2 * root) + 1;  // left child
+    int r = (2 * root) + 2;  // Right child
+    // Check if left child is larger than root.
+    if ((l <= heapsize) && (a[l] > a[root]))
+         largest = l;
+    else
+         largest = root;
```

Comment:
You could initially assign root to largest so that it will reduce this else part.